SConstruct

```diff
@@ -376,23 +384,55 @@ env_base.Prepend(CPPPATH=["#"])
 env_base.platform_exporters = platform_exporters
 env_base.platform_apis = platform_apis
 
-# Build type defines - more platform-specific ones can be in detect.py.
-if env_base["target"] == "release_debug" or env_base["target"] == "debug":
+# Configuration of build targets:
+# - Editor or template
+# - Debug features (DEBUG_ENABLED code)
+# - Dev only code (DEV_ENABLED code)
+# - Optimization level
+# - Debug symbols for crash traces / debuggers
+
+target = env_base["target"]
+env_base.editor_build = target.startswith("editor")
+env_base.dev_build = False
+debug_features = False
+opt_level = ""
+debug_symbols = False
```

Comment:
A bit more of a stretch, but we could also do that here:

```suggestion
debug_features = not target.endswith("release")
opt_level = ""
debug_symbols = not target.endswith("release")
```

By the way, it looks like currently these flags are always set together, even though they mean different things.

Reply:
Yeah or `target != "template-release"`. But same as above, I eventually favored being explicit in the `if`/`elif` that handles each target instead of trying to rely on substrings.

> By the way, it looks like currently these flags are always set together, even though they mean different things.

Indeed, but that's just by chance. `debug_symbols` could really take any value, and is configurable from the command line (and can be stripped anyway after the fact), while `debug_features` is fixed and should not be made configurable (it's basically the only difference between `template-release` and `template-debug`, aside from a different default - but configurable - optimization level).

Reply:
In the end with the new `dev_build` option I refactored this to be much better IMO.